src/goTest.ts

```diff
@@ -274,3 +274,21 @@ export function cleanUpOldAutotestFileOutput() {
 		lastAutotestFileResult = null;
 	}
 }
+
+export function updatePinnedTestLocation(u: vscode.Uri) {
+	if (autorunTestConfig.fileName === u.path) {
```

Comment:
we also should check if this matches vscode.window.activeTextEditor? we only need to update active stuff, not random stuff that changed off-screen

that would also simplify the code below, because you don't need to open the text document, you can use editor.document

Reply:
Yeah I thought about that ... but then we would have to do this work on active editor change? I'll try it that way and see how it feels.

Reply:
oh ya good point

Reply:
Yeah I think I'll keep this the way it is.